packages/material-ui-styles/src/makeStyles.js

```diff
@@ -233,11 +226,14 @@ function makeStyles(stylesOrCreator, options = {}) {
         });
       }
     });
+    React.useEffect(() => {
+      firstRender.current = false;
+    }, []);
 
     // Execute asynchronously every time the theme changes.
     React.useEffect(
       () =>
-        function cleanup() {
+        function componentWillUnmount() {
```

Comment:
Can we move away from the old lifecycle model? It's misleading and in this case (again) wrong. This will cleanup on every theme change not just unmounting.

Reply:
So, we keep cleanup?

> again

I have changed it after seeing Matt pull request, I assumed that you both came up to this wording.

Reply:
I wouldn't even use a named function here. The reason being that it's redundant. Every function you return is a cleanup. Naming this will only cause it to become outdated if you change the dependency argument. I just didn't see code where adding a name here added information to the code.

>I have changed it after seeing Matt pull request, I assumed that you both came up to this wording.

I had the same suggestion there but we kept talking past each other. Don't know if this came to a conclusion. I don't think using a named function here is appropriate for the reasons listed above.

Reply:
Ok, It's fine by me. I'm using an anonymouse function.